src/coreComponents/mesh/generators/HexCellBlockManager.cpp

```diff
@@ -0,0 +1,1126 @@
+
+    LocalFaceIndex idNeighbor = m_allFacesToNeighbors[id];
+    if( idNeighbor != -1)
+    {
+      LocalCellIndex cNeighbor = idNeighbor / Hex::nbFacets;
+      HexFacetIndex faceNeighbor = idNeighbor % Hex::nbFacets;
+      auto blockCellNeighbor = getBlockCellFromManagerCell(cNeighbor) ;
+      getCellBlock( blockCellNeighbor.first ).setElementToFaces( blockCellNeighbor.second, faceNeighbor, curFace );
+    }
+  }
+
+  return true;
+}
+
+
+// Allocation is managed by the CellBlock 
+bool HexMeshConnectivityBuilder::computeElementsToEdgesOfCellBlocks() 
+{
+  for (unsigned int edgeIndex = 0; edgeIndex < m_uniqueEdges.size(); ++edgeIndex)
+  {
+    unsigned int first = m_uniqueEdges[edgeIndex];
+    unsigned int last = m_allEdges.size();
+
+    if( edgeIndex+1 < m_uniqueEdges.size() ) {
+      last = m_uniqueEdges[edgeIndex+1];
+    }
+
+    for(unsigned int i = first ; i < last; ++i)
+    {
+      LocalEdgeIndex id = m_allEdges[i].second;
+      LocalCellIndex c = id / Hex::nbEdges;
+      HexEdgeIndex e = id % Hex::nbEdges;
+      auto blockCell = getBlockCellFromManagerCell(c);
+      getCellBlock(blockCell.first).setElementToEdges(blockCell.second, e, edgeIndex);
+    }
+  }
+  return true;
+}
+
+
+/*********************************************************************************************************************/
+/*********************************************************************************************************************/
+
+HexCellBlockManager::HexCellBlockManager(string const &name, 
+                                         dataRepository::Group *const parent) 
+  :CellBlockManagerBase(name, parent)
+{
+  m_theOneWhoDoesTheJob = nullptr; 
+}
+
+HexCellBlockManager::~HexCellBlockManager()
+{
+  delete m_theOneWhoDoesTheJob;
+  m_theOneWhoDoesTheJob = nullptr;
+}
+
+void HexCellBlockManager::buildMaps()
+{
+  // Create the MeshConnectivityBuilder after the CellBlocks are 
+  // filled and cell types are known
+  // TODO Change the instanciated class depending on the incoming types of cells
+  m_theOneWhoDoesTheJob = new HexMeshConnectivityBuilder( *this );
+
+  // If not called here - the number of faces and edges are not 
+  // available resize EdgeManager and FaceManager.
+  m_theOneWhoDoesTheJob->computeFaces();
+  m_theOneWhoDoesTheJob->computeEdges();
+  
+  m_theOneWhoDoesTheJob->printDebugInformation(); 
+
+  // Otherwise this are not called. Who should?
+  m_theOneWhoDoesTheJob->computeElementsToFacesOfCellBlocks(); 
+  m_theOneWhoDoesTheJob->computeElementsToEdgesOfCellBlocks();
+
+  return ;
+}
+
+localIndex HexCellBlockManager::numEdges() const
+{
+  return m_theOneWhoDoesTheJob->numEdges();
+}
+localIndex HexCellBlockManager::numFaces() const
+{
+  return m_theOneWhoDoesTheJob->numFaces();
+}
+
+array2d<localIndex> HexCellBlockManager::getEdgeToNodes() const
+{
+  array2d<localIndex> result;
+  m_theOneWhoDoesTheJob->computeEdgesToNodes(result);
```

Comment:
So I'm not sure: You compute everything on the fly?
Don't you store anything? could you get some speed up from this?

Reply:
The `MeshConnectivityBuilder` stores information to identify the faces and the edges from 
which all the mapping can be  computed.
```
std::vector< LocalFaceIndex > m_allFacesToNeighbors;  
std::vector< localIndex > m_uniqueFaces;  
std::vector< bool > m_isBoundaryFace;  
std::vector< EdgeInfo > m_allEdges; 
std::vector< localIndex > m_uniqueEdges;
```